src/App.tsx

```diff
@@ -1,203 +1,248 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Status } from './types/Status';
+import { Footer } from './api/Footer';
+import { Header } from './api/Header';
 
-const USER_ID = 0;
+const USER_ID = 136;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setIsError(true))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    switch (status) {
+      case Status.All:
+        getTodos(USER_ID).then(setTodos);
+        break;
+      case Status.Active:
+        setTodos(prevTodos => prevTodos.filter(t => !t.completed));
+        break;
+      case Status.Completed:
+        setTodos(prevTodos => prevTodos.filter(t => t.completed));
+        break;
+      default:
+        break;
+    }
```

Comment:
```suggestion
    switch (status) {
      case Status.All:
        getTodos(USER_ID).then(setTodos);
        break;
        
      case Status.Active:
        setTodos(prevTodos => prevTodos.filter(t => !t.completed));
        break;
        
      case Status.Completed:
        setTodos(prevTodos => prevTodos.filter(t => t.completed));
        break;
        
      default:
        break;
    }
```

Reply:
not fixed 
![image](https://github.com/mate-academy/react_todo-app-loading-todos/assets/70009205/8560c997-c7fa-4a0d-8708-d6e9f6255112)